### PR TITLE
Fix Erlang 24 Warnings

### PR DESCRIPTION
**Pull request by x1001100011**
There're 3 major points for this pr:

### http_uri -> uri_string
`http_uri:decode` -> `emqx_http_lib:uri_decode`
`http_uri:parse` -> `emqx_http_lib:uri_parse`
`http_uri:encode` -> `emqx_http_lib:uri_encode`

### ssl:ssl_accept -> ssl:handshake
https://github.com/erlang/otp/blob/maint-23/lib/ssl/src/ssl.erl#L699

### make test
`All 23 tests passed.` for both master and this pr

### CODE REVIEW

**Review by zmstone — APPROVED** [review]

File: src/er_coap_client.erl [hunk 1]
```diff
@@ -112,7 +120,7 @@ split_segments(Path, Char, Acc) ->
     end.
 
 make_segment(Seg) ->
-    list_to_binary(http_uri:decode(Seg)).
+    list_to_binary(uri_string:percent_decode(Seg)).
```
Comment (zmstone) [inline src/er_coap_client.erl]: If I remember correctly, this is not compatible.

Comment (zmstone) [reply src/er_coap_client.erl]: http://erlang.2086793.n4.nabble.com/http-uri-decode-equivalent-in-uri-string-module-td4727306.html

**Comment by zmstone** [comment]
need a github action for this.

**Review by zmstone — APPROVED** [review]

File: rebar.config [hunk 2]
```diff
@@ -5,6 +5,7 @@
             warn_obsolete_guard,
             debug_info,
             {parse_transform}]}.
+{deps, [{emqx_http_lib, {git, "https://github.com/emqx/emqx_http_lib", {tag, "0.1.0"}}}]}.
```
Comment (x1001100011) [inline rebar.config]:
```suggestion
{deps, [{emqx_http_lib, {git, "https://github.com/emqx/emqx_http_lib", {tag, "0.2.0"}}}]}.
```

**Comment by zmstone** [comment]
tagged: v1.0.1

**Pull request merged**